### PR TITLE
Fix metadata when exporting multiple tracks (#4996)

### PR DESCRIPTION
**Pull request by JohannesLorenz**

### CODE REVIEW

**Comment by JohannesLorenz** [comment]
I'm adding the 1.2.0 label as a reminder. If you want to merge it without review, that's OK. If there's time for a review before, even better.

File: src/core/Mixer.cpp [hunk 1]
```diff
@@ -575,21 +575,35 @@ void Mixer::changeQuality( const struct qualitySettings & _qs )
 
 
 
-void Mixer::setAudioDevice( AudioDevice * _dev,
-			    bool startNow )
+void Mixer::doSetAudioDevice( AudioDevice * _dev )
 {
-	stopProcessing();
+	// TODO: Use shared_ptr here in the future.
```
Comment (PhysSong) [inline src/core/Mixer.cpp]: Why don't you do this now? Also, `QSharedPointer` will work as well as `std::shared_pointer`.

Comment (JohannesLorenz) [reply src/core/Mixer.cpp]: Fixed in fc5cb9c

**Comment by JohannesLorenz** [comment]
Tested, also tested jack. It all seems to work. Let me know if you want to change something, or if it can be merge.

**Comment by PhysSong** [comment]
* I think you shouldn't include translation updates here.
* Windows builds are failing because C++11 is disabled for LADSPA: https://github.com/LMMS/lmms/blob/47d6388182e9105c0cf7e87d494dd2db593350c5/plugins/LadspaEffect/CMakeLists.txt#L5-L8
You can find the pretty old CI errors in https://travis-ci.org/LMMS/lmms/jobs/253835510#L1277-L1289, I'm not sure if we can re-enable it now. If `std::shared_pointer` doesn't work, we should move to `QSharedPointer` or just revert relevant commits.

**Comment by JohannesLorenz** [comment]
Do you propose to not add translation updates to this branch or not to stable-1.2?

About the Windows error: Has this been fixed meanwhile? Maybe the CI now works with C++11? If not, I'd say let's move my shared_ptr fix aside and merge the version from yesterday?

I did try QSharedPointer, but the Qt4 version did not have a proper reset function, I didn't like the workarounds.

**Comment by PhysSong** [comment]
> Do you propose to not add translation updates to this branch or not to stable-1.2?

Merging that commit into `stable-1.2` is okay, but not by this pull request.

> About the Windows error: Has this been fixed meanwhile?

This patch works(MinGW requires `-std=gnu++0x` for `M_PI`):
```diff
diff --git a/plugins/LadspaEffect/CMakeLists.txt b/plugins/LadspaEffect/CMakeLists.txt
index 029cd9168..1c500673c 100644
--- a/plugins/LadspaEffect/CMakeLists.txt
+++ b/plugins/LadspaEffect/CMakeLists.txt
@@ -2,8 +2,10 @@ INCLUDE(BuildPlugin)
 
 # Disable C++11
 REMOVE_DEFINITIONS(-std=c++0x)
-# Enable C++11 for CXXFLAGS only and not for Windows
-IF(NOT LMMS_BUILD_WIN32)
+# Enable C++11 for CXXFLAGS only
+IF("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+	SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++0x")
+ELSE()
 	SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x")
 ENDIF()
```
Anyway, please feel free to defer changes related to shared pointers to `master` if you want to do so.

**Comment by JohannesLorenz** [comment]
Would you prefer to apply that C++11 fix to Ladspa and then release with `shared_ptr`, or only apply `shared_ptr` on `master`? The first option may be a bit cleaner, but it's risky to do that on `stable-1.2`, right?

**Comment by PhysSong** [comment]
If you think that change is unnecessary, you can remove the shared pointer part from here.

**Comment by JohannesLorenz** [comment]
OK, I did an undo of the German translation and and undo of the `shared_ptr`. I'll merge this now and later add the German translation to `stable-1.2` and the `shared_ptr` to `master`.

**Pull request merged**